internal/libyaml/reader_test.go

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package libyaml
+
+import (
+	"testing"
+
+	"go.yaml.in/yaml/v4/internal/testutil/assert"
+)
+
+func TestReader(t *testing.T) {
+	RunTestCases(t, "reader_test.yaml", map[string]TestHandler{
+		"reader-set-error":          runReaderSetErrorTest,
+		"reader-determine-encoding": runReaderDetermineEncodingTest,
+		"reader-update-raw-buffer":  runReaderUpdateRawBufferTest,
+		"reader-update-buffer":      runReaderUpdateBufferTest,
+		"reader-panic":              runReaderPanicTest,
+	})
+}
+
+func runReaderSetErrorTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	parser := NewParser()
+
+	// args should be [problem, offset, value]
+	problem := tc.Args[0].(string)
+	offset := tc.Args[1].(int)
+	value := tc.Args[2].(int)
+
+	result := parser.setReaderError(problem, offset, value)
+
+	// Check return value
+	want := tc.Want.(bool)
+	assert.Equalf(t, want, result, "setReaderError() = %v, want %v", result, want)
+
+	// Run field checks
+	runFieldChecks(t, &parser, tc.Checks)
+}
+
+func runReaderDetermineEncodingTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	parser := NewParser()
+	parser.SetInputString(tc.Input)
+
+	result := parser.determineEncoding()
+
+	// Check return value (defaults to true)
+	want := WantBool(tc.Want, true)
+	assert.Equalf(t, want, result, "determineEncoding() = %v, want %v", result, want)
+
+	// Run field checks
+	runFieldChecks(t, &parser, tc.Checks)
+}
+
+func runReaderUpdateRawBufferTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	parser := NewParser()
+	parser.SetInputString(tc.Input)
+
+	// Apply any setup
+	if tc.Setup != nil {
+		applySetup(t, &parser, tc.Setup)
+	}
+
+	result := parser.updateRawBuffer()
+
+	// Check return value (defaults to true)
+	want := WantBool(tc.Want, true)
+	assert.Equalf(t, want, result, "updateRawBuffer() = %v, want %v", result, want)
+
+	// Run field checks
+	runFieldChecks(t, &parser, tc.Checks)
+}
+
+func runReaderUpdateBufferTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	parser := NewParser()
+	parser.SetInputString(tc.Input)
+
+	// Apply any setup
+	if tc.Setup != nil {
+		applySetup(t, &parser, tc.Setup)
+	}
+
+	// Get the length argument
+	length := tc.Args[0].(int)
```

Comment:
There are too many type assertion for me in this code. Also, a lot of them are not using the ", ok" pattern it sounds a source of panic for anyone who will try to add new tests or update existing one.

Reply:
Fixed in 8c51bbf 3a25a63 76ca2c1 5d077c8